service/matching/deployment_util.go

```diff
@@ -47,3 +48,28 @@ func findDeployment(deployments *persistencespb.DeploymentData, deployment *depl
 	}
 	return -1
 }
+
+func findDeploymentVersion(deployments *persistencespb.DeploymentData, dv *deploymentpb.WorkerDeploymentVersion) int {
+	for i, v := range deployments.GetVersions() {
+		if v.Version.Equal(dv) {
+			return i
+		}
+	}
+	return -1
+}
+
+func hasDeploymentVersion(deployments *persistencespb.DeploymentData, deployment *deploymentpb.Deployment) bool {
```

Comment:
I think it would be nicer if `hasDeploymentVersion` took in a `dv *deploymentpb.WorkerDeploymentVersion` instead of the old `deployment *deploymentpb.Deployment` but it's not a big deal.